src/myrtlespeech/data/preprocess.py

```diff
@@ -222,3 +225,75 @@ def __repr__(self) -> str:
             + f" n_feature_masks={self.n_feature_masks},"
             + f" n_time_masks={self.n_time_masks})"
         )
+
+
+class MFCCLegacy:
+    """Legacy MFCC implementation using ``python_speech_features``.
+
+    This is required for backwards compatibility: **new users should not use
+    this class.** This class has the same API as
+    :py:class`torchaudio.transforms.MFCC`.
+
+    Args:
+        n_mfcc: Number of MFCC coefficients.
+
+        melkwargs: Dict containing mel-spectogram kwargs:
+
+            win_length:
+                Window length (in samples not ms.)
+
+            hop_length:
+                Stride length (in samples not ms.)
+
+        sample_rate: audio sample rate.
+    """
+
+    def __init__(self, n_mfcc, melkwargs, sample_rate=16000):
+        win_length = melkwargs.get("win_length")
+        hop_length = melkwargs.get("hop_length")
+        self.n_mfcc = n_mfcc
+
+        # convert from torchaudio -> python_speech_features equivalents
+        self.samplerate = sample_rate
+        self.numcep = self.n_mfcc
+        self.winlen = win_length / sample_rate
+        self.winstep = hop_length / sample_rate
+
+        # Add normalise/transpose and unsqueeze lambdas
+
+        # deepspeech_internal loader normalises data as it reads it in
+        # Modern PyTorch preserves type when tensors are converted to numpy
+        # but in PyTorch 0.4, the numpy type is always float64, leading
+        # to rounding errors with modern PyTorch where this behaviour is
+        # relied on
+        self.normalise = Lambda(
+            lambda_fn=lambda x: (x * (1 << 15))
+            .to(torch.int16)
+            .squeeze()
+            .numpy()
+            .astype(np.float64)
+        )
+        self.transpose = Lambda(lambda_fn=lambda x: x.transpose(0, 1))
+        self.unsqueeze = Lambda(lambda_fn=lambda x: x.unsqueeze(0))
```

Comment:
Why are these `Lamba`s and not just direct calls in `__call__`?

Reply:
Agreed - these are gone!